applications/gui/modules/text_input.c

```diff
@@ -369,13 +417,28 @@ TextInput* text_input_alloc() {
     view_set_draw_callback(text_input->view, text_input_view_draw_callback);
     view_set_input_callback(text_input->view, text_input_view_input_callback);
 
+    text_input->timer = osTimerNew(text_input_timer_callback, osTimerOnce, text_input, NULL);
+
+    with_view_model(
+        text_input->view, (TextInputModel * model) {
+            string_init(model->validator_text);
+            return false;
+        });
+
     text_input_clean(text_input);
 
     return text_input;
 }
 
 void text_input_free(TextInput* text_input) {
     furi_assert(text_input);
+    with_view_model(
+        text_input->view, (TextInputModel * model) {
+            string_clear(model->validator_text);
+            return false;
+        });
+    osTimerStop(text_input->timer);
+    osTimerDelete(text_input->timer);
```

Comment:
Keep in mind: osTimerDelete doesn't guarantee that timer is actually stoped and deleted.